openadapt/strategies/mixins/layout.py

```diff
@@ -0,0 +1,39 @@
+"""
+Implements a ReplayStrategy mixin for extracting layout and important
+information from text documents and GUI images.
+
+Usage:
+
+    class MyReplayStrategy(LayoutExtractionReplayStrategyMixin):
+        ...
+"""
+from PIL import Image
+from typing import List
+from transformers import pipeline
+import numpy as np
+
+from openadapt.models import Recording, Screenshot, ActionEvent
+from openadapt.strategies.base import BaseReplayStrategy
+
+
+class LayoutExtractionReplayStrategyMixin(BaseReplayStrategy):
+
+    def __init__(self, recording: Recording, image_file_paths: List[str]):
+        super().__init__(recording)
+        self.images = [
+            Image.open(img_file_path).convert("RGB")
+            for img_file_path in image_file_paths
+        ]
+
+    def document_query(self, image: np.ndarray, question: str) -> str:
```

Comment:
Let's move this function to `openadapt/tools/layout.py`. Let's also modify it as follows:
```
def document_query(self, image: np.ndarray | None, image_path: str | None, question: str) -> str:
    assert image or image_path, "At least one of image or image_path must be supplied"
    assert not (image and image_path), "Only one of image or image_path may be supplied"
    if image_path:
        image = Image.open(image_path).convert("RGB")
    ...
```